cmp/grid/Grid.js

```diff
@@ -112,6 +112,7 @@ export class Grid extends Component {
                     'ag-grid-holder',
                     XH.darkTheme ? 'ag-theme-balham-dark' : 'ag-theme-balham',
                     compact ? 'xh-grid-compact' : 'xh-grid-standard',
```

Comment:
See the comment on line 104.  Consulting the stores 'records' property here in render (via this tree getter, or not, see comment below) seems to go directly against this comment, and might yield that 'extra rendering and jumpiness' mentioned.  Perhaps it makes more sense to simply have a private 'isHierarchical' boolean flag on the grid that can be toggled in the dataReaction, and then used here? It can be observable if necessary, but it should be more far more stable than the records property itself.   Alternatively, the getter on the store could be made 'computed' but I think better not to have that getter (see my comment below).